### PR TITLE
feat(query): Support view in system.columns

### LINKED ISSUE #6922: Feature: Columns of views in system.columns table
**Summary**

This will help to work with 3rd party libraries/IDEs.

```sql
CREATE TABLE t1(a INT UNSIGNED, b Varchar)
CREATE VIEW v1 AS SELECT * FROM t1;

SELECT
  name, database, table, type
FROM
  system.columns
WHERE table = 'v1';
```

### PR DESCRIPTION
**Pull request by johnhaxx7**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Ditto

Closes #6922

### CODE REVIEW

**Comment by johnhaxx7** [comment]
@sundy-li PTAL, thanks!

**Review by TCeason — APPROVED** [review]
Thanks for your pr, LGTM. Please let the ci pass.

**Comment by BohuTANG** [comment]
@mergify update

File: src/query/storages/system/src/columns_table.rs [hunk 1]
```diff
@@ -138,7 +143,20 @@ impl ColumnsTable {
...
+                        let mut planner = Planner::new(ctx.clone());
+                        let (plan, _, _) = planner.plan_sql(query).await?;
+                        let schema = infer_table_schema(&plan.schema()).unwrap();
+                        fields = schema.fields().clone();
+                    } else {
+                        return Err(ErrorCode::Internal(
+                            "Logical error, View Table must have a SelectQuery inside.",
+                        ));
+                    }
+                }
```
Comment (RinChanNOWWW) [inline src/query/storages/system/src/columns_table.rs:158]:
```suggestion
                let fields = if table.engine() == VIEW_ENGINE {
                    if let Some(query) = table.options().get(QUERY) {
                        let mut planner = Planner::new(ctx.clone());
                        let (plan, _, _) = planner.plan_sql(query).await?;
                        let schema = infer_table_schema(&plan.schema())?;
                        schema.fields().clone()
                    } else {
                        return Err(ErrorCode::Internal(
                            "Logical error, View Table must have a SelectQuery inside.",
                        ));
                    }
                } else {
                    table.schema().fields().clone()
                };
```

Comment (johnhaxx7) [reply src/query/storages/system/src/columns_table.rs]: Thank you so much for the code snippet! Updated.

**Comment by johnhaxx7** [comment]
Can someone please help me merge this? The failing pipeline `Dev MacOS / build_x86_64_macos(optional)` is a network issue and not related to this change. Or to show me how to re-run it would be super helpful as well. Thanks!

**Comment by TCeason** [comment]
> Can someone please help me merge this? The failing pipeline `Dev MacOS / build_x86_64_macos(optional)` is a network issue and not related to this change. Or to show me how to re-run it would be super helpful as well. Thanks! 

Done

**Pull request merged**